multiraft/multiraft_test.go

```diff
@@ -150,6 +159,22 @@ func (c *testCluster) waitForElection(i int) *EventLeaderElection {
 	}
 }
 
+// elect is a simplified wrapper around triggerElection and waitForElection which
+// waits for the election to complete on all members of a group.
+// TODO(bdarnell): make this work when membership has been changed after creation.
+func (c *testCluster) elect(leaderIndex int, groupID proto.RaftID) {
+	c.triggerElection(leaderIndex, groupID)
+	for _, i := range c.groups[groupID] {
+		el := c.waitForElection(i)
+		if el.NodeID != c.nodes[leaderIndex].nodeID {
+			c.t.Fatalf("wrong leader elected; wanted node %d but got event %v", leaderIndex, el)
```

Comment:
If another node triggers an election concurrently (shouldn't happen in these tests since they use a manual ticker) or if the designated node has fallen behind the other nodes in the group. It doesn't come up in any of the tests we currently have, but I just wanted to make a note that it's not a guaranteed rigged election.